image-customize

```diff
@@ -82,6 +83,114 @@ class InstallAction(ActionBase):
+        else:
+            raise NotImplementedError(f"unknown build platform: {out}")
+
+        # machine_instance.execute(f"{install_command} {package}", timeout=1800)
+
+    def build_deb(machine, vm_tar, version):
+        build_opts = 'nocheck' if BuildAction.quick else ''
+
+        # build source packge
+        machine.execute(f"""
+            set -eu
+            export DEB_BUILD_OPTIONS='{build_opts}'
+            tar -C /var/tmp/build -xf '{vm_tar}'
+            cd "$(ls -d /var/tmp/build/*)"
+            # find and copy debian packaging directory
+            cp -r "$(dirname $(find -path '*/debian/control'))" .
+            # create orig.tar link for building dsc
+            source=$(awk '/^Source: / {{ print $2 }}' debian/control)
+            ln -s '{vm_tar}' ../${{source}}_{version}.orig.tar.xz
+            # fill in the version number
+            sed -i '1 s/([^)-]*/({version}/' debian/changelog
+
+            dpkg-buildpackage -S -us -uc -nc""")
+
+        # build binary packages
+        machine.execute("cd /var/tmp/build; pbuilder build --buildresult . *.dsc", timeout=1800)
+
+        # install packages
+        machine.execute("dpkg -i /var/tmp/build/*.deb")
+
+    def build_rpm(machine, vm_tar, version):
+        mock_opts = ''
+        if BuildAction.verbose:
+            mock_opts += ' --verbose'
+        if BuildAction.quick:
+            mock_opts += ' --nocheck'
+
+        # HACK: mock has *_builddep_command config options for all backends *except* dnf
+        # dnf builddep insists on a cache, which we don't need as the mock is offline, pre-installed,
+        # and single-use only
+        machine.execute('''sed -i '/self.builddep_command =/ s_self.command_"/bin/true"_' '''
+                        '/usr/lib/python*/*-packages/mockbuild/package_manager.py')
+
+        # build source package
+        machine.execute(f'rpmbuild --define "_srcrpmdir /var/tmp/build" -ts "{vm_tar}"')
+
+        # build binary RPMs from srpm
+        machine.execute('chown -R builder:builder /var/tmp/build')
+        machine.execute("su builder -c 'cd /var/tmp/build; mock --no-clean --no-cleanup-after "
+                        f"--offline --resultdir . {mock_opts} --rebuild *.src.rpm'",
+                        timeout=1800)
+
+        # install RPMs
+        machine.execute('packages=$(find /var/tmp/build -name "*.rpm" -not -name "*.src.rpm"); '
+                        f'rpm -U --force --verbose {"--nodigest --nosignature" if BuildAction.quick else ""} $packages')
+
+    def build_arch(machine, vm_tar, version):
+        # unpack source tree's arch packaging directory (PKGBUILD refers to some files)
+        # and set PKGBUILD variables
+        machine.execute(f"""
+            set -eu
+            tar -C /var/tmp/build -xf '{vm_tar}'
+            cd /var/tmp/build/
+            unpackdir="$(ls)"
+            archdir=$(dirname $(find "$unpackdir" -path '*/arch/PKGBUILD'))
```

Comment:
Why not just run `find .` here?

Reply:
Not a big difference indeed, dropped it.